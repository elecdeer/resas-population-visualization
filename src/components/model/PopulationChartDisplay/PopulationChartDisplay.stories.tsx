import { PopulationChartDisplay, PopulationChartDisplayProps} from "./PopulationChartDisplay";

export default {
	title: "ui/PopulationChartDisplay",
	component: PopulationChartDisplay
}

export const Default = (args: PopulationChartDisplayProps) => <PopulationChartDisplay {...args}/>