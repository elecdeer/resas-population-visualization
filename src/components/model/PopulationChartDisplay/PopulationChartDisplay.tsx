import React from "react";
import { PrefectureCheckStates } from "../../../hooks/usePrefecturesState";
import { PrefecturesRes } from "../../../lib/schema/prefecturesResSchema";
import {
  CartesianGrid,
  Legend,
  Line,
  LineChart,
  ResponsiveContainer,
  Tooltip,
  XAxis,
  YAxis,
} from "recharts";
import { usePopulation } from "../../../hooks/usePopulation";
import { useRandomColors } from "../../../hooks/useRandomColors";
export type PopulationChartDisplayProps = {
  prefectures: PrefecturesRes["result"];
  checkState: PrefectureCheckStates;
  showEstimation?: boolean;
};
// import styles from "./PopulationChartDisplay.module.scss";

const numberFormat = new Intl.NumberFormat("en").format;

export const PopulationChartDisplay: React.VFC<PopulationChartDisplayProps> = ({
  prefectures,
  checkState,
  showEstimation = false,
}) => {
  //今回は事前に全ての県のデータを取得する
  const { data: populations } = usePopulation(prefectures);

  const randomColors = useRandomColors(prefectures.length);

  const renderLines = () =>
    prefectures?.map((pref) => (
      <Line
        dataKey={pref.prefCode}
        name={pref.prefName}
        key={pref.prefCode}
        hide={!checkState[pref.prefCode]}
        stroke={randomColors[pref.prefCode]}
        isAnimationActive={false}
      />
    ));

  //scssモジュールだとグローバルなCSSを付けられないのでstyled-jsxでやる
  return (
    <>
      <style global jsx>{`
        //  アクティブでない凡例を隠す
        li.recharts-legend-item.inactive {
          display: none !important;
        }
      `}</style>
      <ResponsiveContainer width={"100%"} height={300}>
        <LineChart data={populations} >
          <CartesianGrid strokeDasharray="3 3" />
          <XAxis
            dataKey="year"
            allowDuplicatedCategory={false}
            tick={{ fontSize: 10 }}
            label={{
              value: "年度",
              position: "insideBottomRight",
              fontSize: 10,
            }}
          />
          <YAxis
            tick={{ fontSize: 10 }}
            label={{
              value: "人口数",
              position: "insideTopLeft",
              fontSize: 10,
              angle: -90,
              dy: 40,
              dx: 0,
            }}
            tickFormatter={(value) => numberFormat(value)}
          />
          <Tooltip />
          <Legend
            width={120}
            layout="vertical"
            verticalAlign="top"
            align="right"
            wrapperStyle={{
              paddingLeft: 20,
            }}
          />
          {renderLines()}
        </LineChart>
      </ResponsiveContainer>
    </>
  );
};
