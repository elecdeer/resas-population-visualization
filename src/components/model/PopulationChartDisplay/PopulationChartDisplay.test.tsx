import { PopulationChartDisplay } from "./PopulationChartDisplay";
import { render } from "@testing-library/react";

test("PopulationChartDisplay Component Snapshot Test", () => {
  const { asFragment } = render(<PopulationChartDisplay />);
  expect(asFragment()).toMatchSnapshot();
});