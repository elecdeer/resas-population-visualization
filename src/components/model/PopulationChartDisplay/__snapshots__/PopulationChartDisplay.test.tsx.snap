// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopulationChartDisplay Component Snapshot Test 1`] = `
<DocumentFragment>
  <style>
    
        //  アクティブでない凡例を隠す
        li.recharts-legend-item.inactive {
          display: none !important;
        }
      
  </style>
  <div
    class="recharts-responsive-container"
    style="width: 100%; height: 300px;"
  />
</DocumentFragment>
`;
